shell/testing/tester_main.cc

```diff
@@ -513,6 +524,84 @@ int RunTester(const flutter::Settings& settings,
   return completion_observer.GetExitCodeForLastError();
 }
 
+#ifdef _WIN32
+#define EXPORTED __declspec(dllexport)
+#else
+#define EXPORTED __attribute__((visibility("default")))
+#endif
+
+extern "C" {
+EXPORTED Dart_Handle LoadLibraryFromKernel(const char* path) {
+  std::shared_ptr<fml::FileMapping> mapping =
+      fml::FileMapping::CreateReadOnly(path);
+  if (!mapping) {
+    FML_LOG(ERROR) << "Failed to load file at \"" << path << "\".";
+    return Dart_Null();
+  }
+  return DartIsolate::LoadLibraryFromKernel(mapping);
+}
+
+EXPORTED Dart_Handle LookupEntryPoint(const char* uri, const char* name) {
+  if (!uri || !name) {
+    return Dart_Null();
+  }
+  return Dart_GetField(Dart_LookupLibrary(Dart_NewStringFromCString(uri)),
+                       Dart_NewStringFromCString(name));
+}
+
+EXPORTED void Spawn(const char* entrypoint, const char* route) {
+  auto shell = g_shell->get();
+  auto spawn_task = [shell, entrypoint, route]() {
+    auto configuration = RunConfiguration::InferFromSettings(
+        shell->GetSettings(), /*io_worker=*/nullptr,
+        /*launch_type=*/IsolateLaunchType::kExistingGroup);
+    configuration.SetEntrypoint(entrypoint);
+
+    Shell::CreateCallback<PlatformView> on_create_platform_view =
+        fml::MakeCopyable([](Shell& shell) mutable {
+          ImpellerVulkanContextHolder impeller_context_holder;
+          return std::make_unique<TesterPlatformView>(
+              shell, shell.GetTaskRunners(),
+              std::move(impeller_context_holder));
+        });
+
+    Shell::CreateCallback<Rasterizer> on_create_rasterizer = [](Shell& shell) {
+      return std::make_unique<Rasterizer>(
+          shell, Rasterizer::MakeGpuImageBehavior::kBitmap);
+    };
+
+    // Spawn a shell, and keep it running until it has no live ports, then
+    // delete it on the platform thread.
+    auto spawned_shell =
+        shell
+            ->Spawn(std::move(configuration), route, on_create_platform_view,
+                    on_create_rasterizer)
+            .release();
+
+    ConfigureShell(spawned_shell);
+
+    fml::TaskRunner::RunNowOrPostTask(
+        spawned_shell->GetTaskRunners().GetUITaskRunner(), [spawned_shell]() {
+          fml::MessageLoop::GetCurrent().AddTaskObserver(
+              reinterpret_cast<intptr_t>(spawned_shell), [spawned_shell]() {
+                if (spawned_shell->EngineHasLivePorts()) {
+                  return;
+                }
+
+                fml::MessageLoop::GetCurrent().RemoveTaskObserver(
+                    reinterpret_cast<intptr_t>(spawned_shell));
+                // Shell must be deleted on the platform task runner.
+                fml::TaskRunner::RunNowOrPostTask(
+                    spawned_shell->GetTaskRunners().GetPlatformTaskRunner(),
+                    [spawned_shell]() { delete spawned_shell; });
+              });
+        });
+  };
+  fml::TaskRunner::RunNowOrPostTask(
```

Comment:
Added a comment. I've also added a latch because I'm realizing that the only thing keeping the shell alive is a receive port that could get closed between now and when the platform task gets scheduled.